### PR TITLE
[1.25] repo: Release v1.25.2

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
i accidently landed this rather than create a PR (needs branch protection) so i immediately reverted it

this is waiting on #25614

File: changelogs/1.23.5.yaml
Comment (adisuissa) [inline changelogs/1.23.5.yaml:1]: Seems to me that this file and the 1.24.3 file should not be part of the release-1.25 branch, but maybe I'm missing something.

**Review by adisuissa — APPROVED** [review]
LGTM, modulo comment that I'm not sure if I have the right context.

File: changelogs/1.23.5.yaml
Comment (phlax) [reply changelogs/1.23.5.yaml:1]: yes - they are included as they are historical versions ot this one

see version histories here https://storage.googleapis.com/envoy-pr/0052d73/docs/version_history/version_history.html

Comment (phlax) [reply changelogs/1.23.5.yaml:1]: i tend to sync these separately now to prevent this confusion but in this case the PR was waiting for the other releases so i just synced with the release commit

**Pull request merged**
